src/parent.rs

```diff
@@ -25,6 +25,11 @@ impl Subclass {
         let mime_type = chunks.next()?;
         let parent_type = chunks.next()?;
 
+        // Consume the leftovers, if any
+        if chunks.next().is_some() {
+            return None;
```

Comment:
The [spec](https://specifications.freedesktop.org/shared-mime-info-spec/shared-mime-info-spec-latest.html) is unclear as to whether the `subclasses` file should discard any extra field; it does say, in the [globs section](https://specifications.freedesktop.org/shared-mime-info-spec/shared-mime-info-spec-latest.html#idm46152099256048) that:

> Implementations should also ignore further fields, so that the syntax of the globs2 file can be extended in the future.

The reference implementation takes everything that comes after the separator as the parent, but does not perform any further validation. Additionally, if the line is:

```
mime/type<space><space>
```

the reference implementation will happily create an entry for `mime/type` with an empty parent.

I'd be fine to be more strict and drop invalid rules.